### ISSUE TITLE
sharli post-link URL dies with "expecting value" message

### ISSUE THREAD

**Issue by mfioretti** [shaarli/python-shaarli-client#33]
Tonight I I installed shaarli v0.97.2 and v0.10.2 side by side on a server configured with nginx, letsencrypt cert and all (except ntp, I will later), just to compare them.

Then I installed shaarli client with python 3.4, following the procedure described here, with pip and a virtualenv:

https://github.com/shaarli/python-shaarli-client/blob/master/docs/user/installation.rst

I sourced .virtualenvs/shaarli/bin/activate then created this .shaarli/shaarli.conf file:
[shaarli]
url = https://example.com (my real URL, of course)
secret = secret taken by shaarli config panel

Now if I run the shaarli client as specified here https://python-shaarli-client.readthedocs.io/en/latest/user/usage.html I get:

(shaarli) [marco@myserver ~]$ shaarli -c .shaarli/shaarli.conf post-link --url https://w3c.github.io/activitypub/
Traceback (most recent call last):
File "/home/pilatort4/.virtualenvs/shaarli/bin/shaarli", line 11, in
sys.exit(main())
File "/home/pilatort4/.virtualenvs/shaarli/lib64/python3.4/site-packages/shaarli_client/main.py", line 65, in main
print(json.dumps(response.json(), sort_keys=True, indent=4))
File "/home/pilatort4/.virtualenvs/shaarli/lib64/python3.4/site-packages/requests/models.py", line 896, in json
return complexjson.loads(self.text, **kwargs)
File "/usr/lib64/python3.4/json/init.py", line 318, in loads
return _default_decoder.decode(s)
File "/usr/lib64/python3.4/json/decoder.py", line 343, in decode
obj, end = self.raw_decode(s, idx=_w(s, 0).end())
File "/usr/lib64/python3.4/json/decoder.py", line 361, in raw_decode
raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)
(shaarli) [marco@myserver ~]$

what is happening? Should I give more/different options, or what?

Thanks

**Comment by mfioretti** [comment]
Also: from the documentation at it is not clear to me how to:

- add a **NEW**  link  specifying several parameters, e.g. description, creation date, tags...
- **EDIT** an existing link

pointers? Thanks

**Comment by mfioretti** [comment]
I have tried to search for the error messages reported above. The page that **seems** to me to offer some clue as to how deal with that issue is this: https://mcgrattan.org/2017/01/24/ordered-json-in-python-with-requests/

but I honestly don't know enough python, especially in virtual environments, to understand if and how I should apply the suggestions of that page to hack the shaarli_client source myself. Any help is greatly appreciated, I really need to make this work, thanks

**Comment by mfioretti** [comment]
after checking the server logs, I have realized that the problem may be there, instead of the client, because there are entries like "/var/www/html/shaarli/api/v1/info" failed (2: No such file or directory)," there. If this is so, then the problem is that it is not intuitive (to me at least) how to "merge" the nginx configuration from the shaarli with the one generated by certbot

**Comment by mfioretti** [comment]
I have realized that the problems above were caused by a wrong configuration of the server. Now it works. fFor the record, the wrong configuration was due to the fact that it is not documented how to "merge" the suggested shaarli modular configuration for nginx to within the nginx config file generated by certbot, on several distributions, and to the fact that at some point I got a configuration that worked from the browser, so it seemed done, but not for the shaarliclient.
